.gitlab-ci.yml

```diff
@@ -27,58 +27,52 @@ builders-and-yarn:
   stage: builders
   image: registry.gitlab.com/magda-data/magda/data61/magda-builder-docker:master
   retry: 1
+  before_script:
+    - yarn install --no-progress --cache-folder .yarn
   cache:
-    key: $CI_JOB_NAME-$CACHE_VERSION
+    key: JS-$JS_CACHE_VERSION
```

Comment:
cache is shared between pipelines and jobs at the project level by default. So `JS-$JS_CACHE_VERSION` key make it always shares cache across stages and between pipeline (until we change the cache version).
This could be an early stage job cache (e.g. build:yarn ) replace a later stage job (e.g. : dockerize:typescript-apis) and may cause issue when we have more than one branch pipeline running?
how about we still use cache to `speed up runs of a given job in subsequent pipelines` (as it's suggested [here](https://docs.gitlab.com/ee/ci/caching/)), i.e. cache vertically --- across pipeline but do not share cache between jobs.
So here, probably a key `$CI_JOB_NAME-$JS_CACHE_VERSION` is what we want?

And `artifcates` use to pass intermediate build results between stages.

Reply:
@t83714 yeah I think I'll revert this whole thing. It was initially to try to make the kubernetes build work better but it actually slows down things on the build box.